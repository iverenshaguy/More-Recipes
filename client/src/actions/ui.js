import { TOGGLE_MODAL, TOGGLE_REVIEW_FORM, CHANGE_RECIPE_FORM_STATE } from './actionTypes';

const toggleModal = modal => ({
  type: TOGGLE_MODAL,
  payload: modal || null
});


const toggleReviewForm = () => ({
  type: TOGGLE_REVIEW_FORM
});

const changeRecipeFormState = payload => ({
  type: CHANGE_RECIPE_FORM_STATE,
  payload
});

export default {
  toggleModal,
  toggleReviewForm,
  changeRecipeFormState
};
