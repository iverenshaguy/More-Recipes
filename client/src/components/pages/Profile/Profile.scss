@import '../../../assets/scss/modules';

.user-profile {
  background-color: $background-grey !important;

  .navbar {
    background-color: $white;

    a, i, .fa-heart {
      color: $navbar-grey !important;
    }
  }

  .navbar-brand {
    color: $navbar-grey !important;
  }

  #search {
    border: 1px solid $navbar-grey !important;
    border-right: none !important;
  }

  #search-form .input-group-btn button {
    border-color: $navbar-grey !important;
    border-left: none !important;
  }

  #more-recipes-navbar.affix .fa-heart {
    color: $navbar-grey !important;
  }
}

.black {
  color: $darkgrey;
}

.user-profile {
  .card-img-top {
    height: 8rem !important;
    width: 100%;
  }

  .card-block {
    padding-top: 0.5em !important;
    padding-bottom: 0.5em !important;
  }

  .card-top {
    height: 10rem !important;
    color: $navbar-grey;
  }

  #add-recipe-card i {
    padding-top: 23%;
  }
}

.user-info {
  .name {
    font-size: 1.2rem;
  }

  .profile-picture {
    border-right: 1px solid $recipe-card-grey;
  }

  .profile-picture {
    img {
      width: 150px;
      height: 150px;
    }
  }

  .name-div {
    border-right: 0px solid $recipe-card-grey;
  }

  .username, .about-me {
    font-size: 1.1rem;
  }
}

@media only screen and (max-width: 768px) and (min-width: 576px) {
  .user-info {
    .profile-picture img {
      img {
        height: 100px;
        width: 100px;
      }
    }
  }
}

@media only screen and (max-width: 576px) {
  .user-info {
    .profile-picture {
        img {
        height: 90px;
        width: 90px;
      }
    }
  }
}

.user-profile-cards {
  margin-bottom: 15vh;
}

.user-profile-recipe-cards-wrapper {
  margin-top: -3rem;
}

.add-text {
  font-size: 1.2em;
  font-weight: 600;
  color: $darkgrey;
}

.category-card p {
  padding-top: 1em;
  padding-bottom: 1em;
  font-size: 1.2em;
  font-weight: 600;
}

.user-info .recipe-count p {
  font-size: 1.2em;
  text-transform: uppercase;
}

.user-dashboard-pills {
  padding: 2em 5em 0.5em;
}

.nav-pills .nav-link.active {
  background-color: $user-navpills-grey;
  border-radius: 20px;
  color: $darkgrey;
}

.settings-collapse .heading a {
  border-radius: 5px;
  color: $darkgrey;

  /* font-size: 1em; */
  /* font-weight: bold; */
  padding: .4em 1em .5em;
  text-transform: uppercase;
}

.nav-pills .nav-link {
  font-size: 1.2em;
  font-weight: bold;
  padding: .4em 1em .5em;
  text-transform: uppercase;
}

.nav-profile-picture-div {
  height: 2rem;
  width: 2rem;
  text-transform: capitalize;
  border-radius: 50%;

  /*margin-top: -0.06rem;*/

  img {
    height: 2rem;
    width: 2rem;
    text-transform: capitalize;
    border-radius: 50%;
    margin-top: -0.3rem;
  }
}