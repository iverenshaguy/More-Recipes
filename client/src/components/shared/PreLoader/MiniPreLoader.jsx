import React from 'react';
import Loader from 'react-loaders';
import 'loaders.css/src/animations/ball-beat.scss';
import './PreLoader.scss';

const MiniPreLoader = () => (<Loader type="ball-beat" />);

export default MiniPreLoader;
