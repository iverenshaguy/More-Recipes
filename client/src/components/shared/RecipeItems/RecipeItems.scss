@import '../../../assets/scss/modules';

#search-result-wrapper {
  hr {
    border-top-color: $border-top-grey;
    margin-top: 2rem;
    width: 30%;
  }
}

#search-result {
  #side-sorter {
    border-right: 1px solid $light-grey;
  }
}
