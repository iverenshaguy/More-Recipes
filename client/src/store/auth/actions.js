import {
  AUTHENTICATED,
  UNAUTHENTICATED,
  AUTHENTICATION_ERROR,
  AUTHENTICATING,
  LOGIN_SUCCESS,
  LOGIN_ERROR,
  SIGNUP_SUCCESS,
  SIGNUP_ERROR,
  CLEAR_AUTH_ERROR
} from './types';

const authenticating = () => ({
  type: AUTHENTICATING
});

const authenticationSuccess = data => ({
  type: AUTHENTICATED,
  payload: data
});

const authenticationFailure = error => ({
  type: AUTHENTICATION_ERROR,
  payload: error
});

const loginSuccess = data => ({
  type: LOGIN_SUCCESS,
  payload: data
});

const loginFailure = error => ({
  type: LOGIN_ERROR,
  payload: error
});

const signupSuccess = data => ({
  type: SIGNUP_SUCCESS,
  payload: data
});

const signupFailure = error => ({
  type: SIGNUP_ERROR,
  payload: error
});

const clearAuthError = () => ({
  type: CLEAR_AUTH_ERROR
});

const resetUser = () => ({
  type: UNAUTHENTICATED
});

export default {
  authenticating,
  authenticationSuccess,
  authenticationFailure,
  loginSuccess,
  loginFailure,
  signupSuccess,
  signupFailure,
  clearAuthError,
  resetUser
};
