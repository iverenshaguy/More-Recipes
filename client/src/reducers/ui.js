import { TOGGLE_MODAL, TOGGLE_REVIEW_FORM, CHANGE_RECIPE_FORM_STATE } from '../actions/actionTypes';

const initialState = {
  modals: {
    isOpen: false,
    type: null
  },
  reviewForm: {
    isOpen: false
  },
  recipeForm: {
    active: 'one'
  }
};

export default (state = initialState, action) => {
  switch (action.type) {
    case TOGGLE_MODAL:
      return Object.assign({}, state, {
        modals: {
          isOpen: !state.modals.isOpen,
          type: action.payload || null
        }
      });
    case TOGGLE_REVIEW_FORM:
      return Object.assign({}, state, {
        reviewForm: {
          isOpen: !state.reviewForm.isOpen,
        }
      });
    case CHANGE_RECIPE_FORM_STATE:
      return Object.assign({}, state, {
        recipeForm: {
          active: action.payload
        }
      });
    default:
      return state;
  }
};
