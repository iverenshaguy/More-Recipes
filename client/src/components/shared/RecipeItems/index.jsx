import React from 'react';
import PropTypes from 'prop-types';
import RecipeCard from '../RecipeCard';
import Pagination from '../Pagination';
import getNoResultTitle from '../../../helpers/getNoResultTitle';
import './RecipeItems.scss';

/**
 * @exports
 * @function RecipeItems
 * @param {object} props
 * @returns {component} RecipeItems
 */
const RecipeItems = props => (
  <div className="container-fluid pb-5 background-grey page-margin" id="search-result-wrapper">
    <div className="heading pt-4 text-center" id="search-result-heading">
      <div className="col-xs-12 col-md-12 pt-3" id="title">
        <h5>{props.title}</h5>
      </div>
    </div>
    <hr />
    <div className="row pt-2 px-5" id="search-result">
      {props.recipes.length === 0 &&
        <div className="col">
          {props.recipes.length === 0 && <div className="text-center pt-4">{getNoResultTitle(props.title)}</div>}
        </div>}
      {props.recipes.length !== 0 &&
        props.recipes.map(recipe => (<RecipeCard key={recipe.id} recipe={recipe} />))}
    </div>
    <Pagination
      items={props.recipes}
      onPageChange={props.handlePageChange}
      metaData={props.metaData}
    />
  </div>);

RecipeItems.propTypes = {
  title: PropTypes.string.isRequired,
  recipes: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    recipeName: PropTypes.string,
    recipeImage: PropTypes.string,
    prepTime: PropTypes.string,
    cookTime: PropTypes.string,
    totalTime: PropTypes.string,
    difficulty: PropTypes.string,
    extraInfo: PropTypes.string,
    vegetarian: PropTypes.bool,
    ingredients: PropTypes.array,
    preparations: PropTypes.array,
    directions: PropTypes.array,
    upvotes: PropTypes.number,
    downvotes: PropTypes.number,
    views: PropTypes.number,
    createdAt: PropTypes.string,
    updatedAt: PropTypes.string,
    userId: PropTypes.number,
    rating: PropTypes.string
  })).isRequired,
  metaData: PropTypes.shape({
    firstPage: PropTypes.number,
    lastPage: PropTypes.number,
    page: PropTypes.number,
    pageRecipeCount: PropTypes.number,
    pages: PropTypes.arrayOf(PropTypes.number),
    totalRecipeCount: PropTypes.number,
  }).isRequired,
  handlePageChange: PropTypes.func.isRequired
};

export default RecipeItems;
