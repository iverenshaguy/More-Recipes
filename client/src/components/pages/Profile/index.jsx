import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Link } from 'react-router-dom';
import { push } from 'react-router-redux';
import { Button } from 'reactstrap';
import PropTypes from 'prop-types';
import FontAwesome from 'react-fontawesome';
import ProfilePic from './ProfilePic';
import RecipeItems from '../../shared/RecipeItems';
import { MiniPreLoader } from '../../shared/PreLoader';
import { toggleModal } from '../../../actions/ui';
import { fetchUserRecipes } from '../../../actions/recipes';
import { updateUserImage } from '../../../actions/auth';
import {
  setUploading, unsetUploading, uploadSuccess, uploadFailure, clearUploadError, uploadImage
} from '../../../actions/uploadImage';
import { userPropTypes, multiRecipePropTypes, urlMatchPropTypes } from '../../../helpers/proptypes';
import './Profile.scss';

/**
 * @exports
 * @class Profile
 * @extends Component
 * @classdesc Returns User Profile Page
 * @returns {JSX} User Profile
 */
class Profile extends Component {
  static propTypes = {
    ...userPropTypes,
    ...urlMatchPropTypes,
    ...multiRecipePropTypes,
    isFetching: PropTypes.bool.isRequired,
    updateUserImage: PropTypes.func.isRequired
  }

  /**
   * @memberof Profile
   * @constructor
   * @returns {JSX} Profile
   */
  constructor() {
    super();

    this.state = {
      limit: 5,
      currentPage: 1
    };

    this.handlePageChange = this.handlePageChange.bind(this);
    this.showAddRecipeModal = this.showAddRecipeModal.bind(this);
  }

  /**
   * @memberof Profile
   * @returns {nothing} returns nothing
   */
  componentWillMount() {
    const { match, user, dispatch } = this.props;
    if (match.params.username !== user.username) {
      dispatch(push(`/${user.username}`));
    }

    dispatch(fetchUserRecipes(this.props.user.id, this.state.currentPage, this.state.limit));
  }

  /**
   * @memberof Home
   * @param {number} page
   * @returns {state} home
   */
  handlePageChange(page) {
    this.setState({
      currentPage: page
    }, () => this.props.dispatch(fetchUserRecipes(this.props.user.id, page, this.state.limit)));
  }

  /**
   * @memberof Profile
   * @returns {nothing} Returns nothing
   */
  showAddRecipeModal() {
    this.props.dispatch(toggleModal('addRecipe'));
  }

  /**
   * @memberof Profile
   * @returns {JSX} User Profile
   */
  render() {
    const {
      user, isFetching, recipes, metadata, uploadImageObj
    } = this.props;

    return (
      <div className="user-profile pb-4">
        <div className="container-fluid user-profile-div">
          <div className="row justify-content-start user-info py-4 px-3 px-md-5">
            <ProfilePic
              user={user}
              uploadImageObj={uploadImageObj}
              uploadImage={this.props.uploadImage}
              setUploading={this.props.setUploading}
              uploadFailure={this.props.uploadFailure}
              uploadSuccess={this.props.uploadSuccess}
              unsetUploading={this.props.unsetUploading}
              updateUserImage={this.props.updateUserImage}
              clearUploadError={this.props.clearUploadError}
            />
            <div className="col-7 col-md-8 name-div align-self-center">
              <p className="name pt-2">{`${user.firstname} ${user.lastname}`}</p>
              <small className="text-muted username pt-1">{`@${user.username}`}</small>
              <p className="about-me pt-2">{user.aboutMe}</p>
            </div>
          </div>
        </div>
        <div className="container-fluid text-center user-profile-recipe-cards-wrapper mb-5 mt-1">
          <Button
            onClick={this.showAddRecipeModal}
            className="btn-default btn-lg d-none d-md-inline-block"
            id="home-add-recipe-btn"
            title="New Recipe"
          >
            Add a New Recipe
          </Button>
          <div className="profile-toggle-wrapper text-center mt-3">
            <a className="d-inline recipes-tab text-center active" href="#recipes">
              <i className="aria-hidden flaticon flaticon-flat-plate-with-hot-food-from-side-view" />
              <span className="d-none d-sm-inline">&nbsp;&nbsp;RECIPES</span>
            </a>
            <Link to={`/${user.username}/favorites`} className="d-inline recipes-tab text-center" href="#favorite-recipes">
              <FontAwesome name="heart" size="lg" tag="i" />
              <span className="d-none d-sm-inline">&nbsp;&nbsp;FAVORITES</span>
            </Link>
          </div>
          {isFetching &&
            <div className="justify-content-center py-5">
              <MiniPreLoader />
            </div>}
          {!isFetching &&
            <div className="row justify-content-center user-profile-recipe-cards px-5 px-sm-5 px-md-0 mt-0">
              <RecipeItems
                title="MY RECIPES"
                recipes={recipes}
                handlePageChange={this.handlePageChange}
                metadata={metadata}
              />
            </div>}
        </div>
      </div>);
  }
}

const mapStateToProps = state => ({
  user: state.auth.user,
  isFetching: state.isFetching,
  recipes: state.recipes.items,
  metadata: state.recipes.metadata,
  uploadImageObj: state.uploadImage
});

const mapDispatchToProps = dispatch => bindActionCreators({
  clearUploadError,
  updateUserImage,
  unsetUploading,
  uploadSuccess,
  uploadFailure,
  setUploading,
  uploadImage
}, dispatch);

export { Profile as ProfileComponent };
export default connect(mapStateToProps, mapDispatchToProps)(Profile);
