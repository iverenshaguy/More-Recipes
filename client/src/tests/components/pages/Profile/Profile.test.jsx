/* eslint-disable */
import React from 'react';
import { toggleModal } from '../../../../actions/ui';
import RecipeItems from '../../../../components/shared/RecipeItems';
import MiniPreLoader from '../../../../components/shared/PreLoader/MiniPreLoader';
import { ProfileComponent } from '../../../../components/pages/Profile';

const mockFn = jest.fn();
const setup = () => {
  const props = {
    dispatch: mockFn,
    isFetching: false,
    match: {
      url: '/',
      params: {
        username: 'username'
      }
    },
    user: {
      id: 2,
      username: 'username'
    },
    recipes: [{ id: 1, recipeName: 'Rice', User: 'Me' }, { id: 5, recipeName: 'Beans', User: 'She' }],
    metadata: {
      firstPage: 1,
      lastPage: 3,
      page: 2,
      itemsPerPage: 5,
      pages: [1, 2, 3],
      totalCount: 13,
    },
    uploadImage: {
      uploading: false,
      success: false
    }
  };

  const shallowWrapper = shallow(<ProfileComponent {...props} />, rrcMock.get());
  const mountedWrapper = mount(<ProfileComponent {...props} />, rrcMock.get());

  return { props, shallowWrapper, mountedWrapper };
};

const state = {
  limit: 1,
  uploadTask: {
    delete: jest.fn(),
    cancel: jest.fn()
  },
  currentPage: 1,
  uploading: false,
  uploadError: null
};

describe('Profile', () => {
  it('renders correctly', () => {
    const { shallowWrapper } = setup();
    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('calls componentWillMount and dispatch when mounted', () => {
    const { mountedWrapper } = setup();

    expect(mockFn).toHaveBeenCalled();
    mountedWrapper.unmount();
  });

  it('renders Redirects to user\'s profile params is not equal to username', () => {
    const { props } = setup();
    // rrcMock = global variable that mocks react-router context
    const shallowWrapper = shallow(<ProfileComponent
      {...props}
      user={{ id: 1, username: 'username2' }}
    />, rrcMock.get());

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('renders user\'s RecipeItmes when isFetching is false', () => {
    const { shallowWrapper } = setup();

    expect(shallowWrapper.find(RecipeItems).length).toBeTruthy();
  });

  it('renders MiniPreloader when isFetching is true', () => {
    const { props } = setup();
    const shallowWrapper = shallow(<ProfileComponent {...props} isFetching />, rrcMock.get());

    expect(shallowWrapper.find(MiniPreLoader).length).toBeTruthy();
  });

  it('renders error when uploading is false and there is an error', () => {
    const { shallowWrapper } = setup();

    shallowWrapper.setState({ uploading: false, uploadError: 'An error occurred' });

    expect(shallowWrapper.find('.upload-error').length).toBeTruthy();
    expect(shallowWrapper.find('.upload-error').text()).toEqual('An error occurred');
  });

  it('renders change picture link and changes image opacity when not uploading', () => {
    const { shallowWrapper } = setup();

    shallowWrapper.setState({ uploading: false });
    const wrapperStyle = shallowWrapper.find('img').get(0).props.style;

    expect(shallowWrapper.find('a[href="#profile-photo"]').text()).toEqual('Change Picture');
    expect(wrapperStyle).toHaveProperty('opacity', 1);
  });

  it('renders cancel upload link and changes image opacity when uploading', () => {
    const { shallowWrapper } = setup();

    shallowWrapper.setState({ uploading: true });
    const wrapperStyle = shallowWrapper.find('img').get(0).props.style;

    expect(shallowWrapper.find('a[href="#profile-photo"]').text()).toEqual('Cancel Upload');
    expect(wrapperStyle).toHaveProperty('opacity', 0.3);
  });

  it('renders user\'s profilepic when provided', () => {
    const { props } = setup();
    const shallowWrapper = shallow(<ProfileComponent
      {...props}
      user={{ id: 1, username: 'username', profilePic: 'pic.png' }}
    />, rrcMock.get());

    shallowWrapper.setState({ uploading: true });

    expect(shallowWrapper.find('img[src="pic.png"]')).toBeTruthy();
  });

  it('handles page change', () => {
    const { mountedWrapper } = setup();

    const handlePageChangeSpy = jest.spyOn(mountedWrapper.instance(), 'handlePageChange');

    mountedWrapper.setState(state);
    mountedWrapper.find('a.page-link').at(3).simulate('click');

    expect(handlePageChangeSpy).toHaveBeenCalled();
    expect(mountedWrapper.instance().state.currentPage).toEqual(2);
    mountedWrapper.unmount();
  });

  it('calls showAddRecipeModal() when button is clicked', () => {
    const { mountedWrapper } = setup();
    const showAddRecipeModalSpy = jest.spyOn(mountedWrapper.instance(), 'showAddRecipeModal');

    mountedWrapper.setState(state);
    mountedWrapper.find('button#home-add-recipe-btn').simulate('click');

    expect(showAddRecipeModalSpy).toHaveBeenCalled();
    expect(mockFn).toHaveBeenCalledWith(toggleModal('addRecipe'));
    mountedWrapper.unmount();
  });

  it('handles change image click', () => {
    const { mountedWrapper } = setup();
    const handleCancelImageUploadSpy = jest.spyOn(mountedWrapper.instance(), 'handleCancelImageUpload');
    const mockClick = jest.fn();

    mountedWrapper.instance().imageUploader.click = mockClick;
    mountedWrapper.setState({ ...state, uploadError: 'Error' });
    mountedWrapper.find('a[href="#profile-photo"]').simulate('click');

    expect(mountedWrapper.instance().state.uploadError).toBeNull();
    expect(handleCancelImageUploadSpy).not.toHaveBeenCalled();
    expect(mockClick).toHaveBeenCalled();
    mountedWrapper.unmount();
  });

  it('handles cancel upload click', () => {
    const { mountedWrapper } = setup();
    const handleCancelImageUploadSpy = jest.spyOn(mountedWrapper.instance(), 'handleCancelImageUpload');
    const mockClick = jest.fn();

    mountedWrapper.instance().imageUploader.click = mockClick;
    mountedWrapper.setState({ ...state, uploadError: 'Error', uploading: true });
    mountedWrapper.find('a[href="#profile-photo"]').simulate('click');
    mountedWrapper.find('img').src = 'picn';

    expect(mountedWrapper.instance().state.uploadError).toBeNull();
    expect(mountedWrapper.find('img[src="images/user-image-placeholder.png"]')).toBeTruthy();
    expect(handleCancelImageUploadSpy).toHaveBeenCalled();
    expect(mockClick).not.toHaveBeenCalled();
    mountedWrapper.unmount();
  });

  it('handles change image', () => {
    const { mountedWrapper } = setup();
    const handleChangeImageSpy = jest.spyOn(mountedWrapper.instance(), 'handleChangeImage');
    const image = new Blob(['../../assets/images/user-image-placeholder.png'], { type: 'application/png' });
    const event = { target: { files: [image] } };

    mountedWrapper.setState(state);
    mountedWrapper.find('input[type="file"]').simulate('change', event);

    expect(handleChangeImageSpy).toHaveBeenCalled();
    mountedWrapper.unmount();
  });

  // it('handles image upload: successful upload', () => {
  //   const { props, mountedWrapper } = setup();
  //   const handleImageUploadSpy = jest.spyOn(mountedWrapper.instance(), 'handleImageUpload');
  //   const image = new Blob(['../../assets/images/user-image-placeholder.png'], { type: 'application/png', size: 2 * 1024 });
  //   const event = { target: { files: [image] } };

  //   mountedWrapper.setState(state);
  //   mountedWrapper.find('input[type="file"]').simulate('change', event);
  //   mountedWrapper.setProps({ ...props, uploadImage: { uploading: false, uploadSuccess: true } });

  //   expect(handleImageUploadSpy).toHaveBeenCalled();
  //   expect(mountedWrapper.instance().state.uploading).toBeFalsy();
  //   mountedWrapper.unmount();
  // });

  // it('handles image upload: unsuccessful upload', () => {
  //   const { props, mountedWrapper } = setup();
  //   const handleImageUploadSpy = jest.spyOn(mountedWrapper.instance(), 'handleImageUpload');
  //   const image = new Blob(['../../assets/images/user-image-placeholder.png'], { type: 'application/png', size: 2 * 1024 });
  //   const event = { target: { files: [image] } };

  //   mountedWrapper.setState(state);
  //   mountedWrapper.find('input[type="file"]').simulate('change', event);
  //   mountedWrapper.setProps({ ...props, uploadImage: { uploading: false, uploadSuccess: false } });

  //   expect(handleImageUploadSpy).toHaveBeenCalled();
  //   expect(mountedWrapper.instance().state.uploading).toBeFalsy();
  //   expect(mountedWrapper.instance().state.uploadError).toEqual('Something happened, please try again');
  //   expect(props.uploadTask.delete).toHaveBeenCalled();
  //   mountedWrapper.unmount();
  // });
});
