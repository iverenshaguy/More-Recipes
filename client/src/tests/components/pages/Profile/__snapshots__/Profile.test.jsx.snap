// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile renders Redirects to user's profile params is not equal to username 1`] = `
<div
  className="user-profile pb-4"
>
  <div
    className="container-fluid user-profile-div"
  >
    <div
      className="row justify-content-start user-info py-4 px-3 px-md-5"
    >
      <div
        className="col-5 col-md-4 profile-picture align-self-center"
      >
        <img
          alt="Profile"
          className="d-block d-md-inline rounded-circle img-fluid img-thumbnail user-img align-middle"
          src="images/user-image-placeholder.png"
          style={
            Object {
              "opacity": 1,
            }
          }
        />
           
        <div
          className="d-block d-md-inline-block align-middle"
        >
          <a
            href="#profile-photo"
            onClick={[Function]}
          >
            Change Picture
          </a>
        </div>
        <input
          accept="image/gif, image/jpeg, image/png"
          onChange={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
          type="file"
        />
      </div>
      <div
        className="col-7 col-md-8 name-div align-self-center"
      >
        <p
          className="name pt-2"
        >
          undefined undefined
        </p>
        <small
          className="text-muted username pt-1"
        >
          @username2
        </small>
        <p
          className="about-me pt-2"
        />
      </div>
    </div>
  </div>
  <div
    className="container-fluid text-center user-profile-recipe-cards-wrapper mb-5 mt-1"
  >
    <Button
      className="btn-default btn-lg d-none d-md-inline-block"
      color="secondary"
      id="home-add-recipe-btn"
      onClick={[Function]}
      tag="button"
      title="New Recipe"
    >
      Add a New Recipe
    </Button>
    <div
      className="row justify-content-center user-profile-recipe-cards px-5 px-sm-5 px-md-0 mt-0"
    >
      <RecipeItems
        handlePageChange={[Function]}
        metadata={
          Object {
            "firstPage": 1,
            "itemsPerPage": 5,
            "lastPage": 3,
            "page": 2,
            "pages": Array [
              1,
              2,
              3,
            ],
            "totalCount": 13,
          }
        }
        recipes={
          Array [
            Object {
              "User": "Me",
              "id": 1,
              "recipeName": "Rice",
            },
            Object {
              "User": "She",
              "id": 5,
              "recipeName": "Beans",
            },
          ]
        }
        title="MY RECIPES"
      />
    </div>
  </div>
</div>
`;

exports[`Profile renders correctly 1`] = `
<div
  className="user-profile pb-4"
>
  <div
    className="container-fluid user-profile-div"
  >
    <div
      className="row justify-content-start user-info py-4 px-3 px-md-5"
    >
      <div
        className="col-5 col-md-4 profile-picture align-self-center"
      >
        <img
          alt="Profile"
          className="d-block d-md-inline rounded-circle img-fluid img-thumbnail user-img align-middle"
          src="images/user-image-placeholder.png"
          style={
            Object {
              "opacity": 1,
            }
          }
        />
           
        <div
          className="d-block d-md-inline-block align-middle"
        >
          <a
            href="#profile-photo"
            onClick={[Function]}
          >
            Change Picture
          </a>
        </div>
        <input
          accept="image/gif, image/jpeg, image/png"
          onChange={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
          type="file"
        />
      </div>
      <div
        className="col-7 col-md-8 name-div align-self-center"
      >
        <p
          className="name pt-2"
        >
          undefined undefined
        </p>
        <small
          className="text-muted username pt-1"
        >
          @username
        </small>
        <p
          className="about-me pt-2"
        />
      </div>
    </div>
  </div>
  <div
    className="container-fluid text-center user-profile-recipe-cards-wrapper mb-5 mt-1"
  >
    <Button
      className="btn-default btn-lg d-none d-md-inline-block"
      color="secondary"
      id="home-add-recipe-btn"
      onClick={[Function]}
      tag="button"
      title="New Recipe"
    >
      Add a New Recipe
    </Button>
    <div
      className="row justify-content-center user-profile-recipe-cards px-5 px-sm-5 px-md-0 mt-0"
    >
      <RecipeItems
        handlePageChange={[Function]}
        metadata={
          Object {
            "firstPage": 1,
            "itemsPerPage": 5,
            "lastPage": 3,
            "page": 2,
            "pages": Array [
              1,
              2,
              3,
            ],
            "totalCount": 13,
          }
        }
        recipes={
          Array [
            Object {
              "User": "Me",
              "id": 1,
              "recipeName": "Rice",
            },
            Object {
              "User": "She",
              "id": 5,
              "recipeName": "Beans",
            },
          ]
        }
        title="MY RECIPES"
      />
    </div>
  </div>
</div>
`;
